lib/modem_trace/modem_trace.c

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <kernel.h>
+#include <zephyr.h>
+#include <modem/modem_trace.h>
+#include "nrf_modem_at.h"
+#include "nrfx_uarte.h"
+#include "SEGGER_RTT.h"
+
+static const nrfx_uarte_t uarte_inst = NRFX_UARTE_INSTANCE(1);
+
+static int trace_rtt_channel;
+static char rtt_buffer[CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_RTT_BUF_SIZE];
+
+static uint32_t max_trace_size_bytes;
+static uint32_t total_trace_size_rcvd;
+static bool is_transport_initialized;
+
+static void trace_stop_timer_handler(struct k_timer *timer);
+
+K_TIMER_DEFINE(trace_stop_timer, trace_stop_timer_handler, NULL);
+
+static void trace_stop_timer_handler(struct k_timer *timer)
+{
+	modem_trace_stop();
+}
+
+static bool uart_init(void)
+{
+	const nrfx_uarte_config_t config = {
+		.pseltxd = DT_PROP(DT_NODELABEL(uart1), tx_pin),
+		.pselrxd = DT_PROP(DT_NODELABEL(uart1), rx_pin),
+		.pselcts = NRF_UARTE_PSEL_DISCONNECTED,
+		.pselrts = NRF_UARTE_PSEL_DISCONNECTED,
```

Comment:
this code does not implement a device for the nrf UART/E compatible, so why using this information?

Reply:
Not sure what you mean. Could you please elaborate? Also this is not new code. I just moved old code from nrf_modem_os.c to this place.